internal/lsp/completions/providers/package.go

```diff
@@ -0,0 +1,64 @@
+package providers
+
+import (
+	"strings"
+
+	"github.com/styrainc/regal/internal/lsp/cache"
+	"github.com/styrainc/regal/internal/lsp/types"
+)
+
+// Package will return completions for the package keyword when starting a new file.
+type Package struct{}
+
+func (*Package) Run(c *cache.Cache, params types.CompletionParams) ([]types.CompletionItem, error) {
+	fileURI := params.TextDocument.URI
+
+	fileContents, ok := c.GetFileContents(fileURI)
+	if !ok {
+		// if the file contents is missing then we can't provide completions
+		return []types.CompletionItem{}, nil
+	}
+
+	lines := strings.Split(fileContents, "\n")
+	if len(lines) < 1 {
+		return []types.CompletionItem{}, nil
+	}
```

Comment:
Looks like pretty much all of the `Run` implementations share the first 10 lines, which makes sense as they'll all need the contents split by line. I wonder if it would make sense to have the manager do this instead and have it passed to the `Run` functions. Not so much as an optimization (but it won't hurt) but just to avoid cluttering each of the `Run` implementations with the same setup code.

Reply:
Very open to revisiting when we see what shape the other providers start to look like.